docs/development.md

```diff
@@ -0,0 +1,420 @@
+## How to develop a Filter
+
+A filter is part of a route and can change arbitary http data in the
+`http.Request` and `http.Response` path of a proxy.
+
+The filter example shows a non trivial diff of a filter
+implementation, that implements an authnz webhook. It shows global
+settings passed via flags, user documentation, developer documentation
+for library users, the filter implementation and some test
+cases. Tests should test the actual filter implementation in a proxy
+setup.
+
+### How to pass options to your filter
+
+Set a `default` and a `Usage` string as const.  Add a var to hold the
+value and put the flag to the category, that makes the most sense.
+
+If a filter, predicate or dataclient need `Options` passed from flags,
+then you should register the filter in `skipper.go`, the main library
+entrypoint. In case you do not need options from flags, use
+`MakeRegistry()` in `./filters/builtin/builtin.go` to register your
+filter.
+
+
+```
+diff --git a/cmd/skipper/main.go b/cmd/skipper/main.go
```

Comment:
i personally really like the code diff in the docs, never seen something like that before :+1: